### PR TITLE
EZP-25041 implement getField method on User and UserGroup

### PR DESCRIPTION
**Pull request by iherak**
Issue: https://jira.ez.no/browse/EZP-25041

User and UserGroup do not implement getField method, which in turn means they can not be used with translation and field helper, nor can the ez_field_value and ez_render_field be used directly on User/UserGroup (only on their 'content' property, accessed via magic method).
As they can be used however with ez_render_name, for example, and considering they do extend abstract 'content' class, it would make sense for them to behave in the same way the Content itself does.

This PR moves getField method to abstract class Content and implements it in User and UserGroup.

### CODE REVIEW

**Comment by iherak** [comment]
The test failed on ContentTypeServiceTest - I do not think it should have anything to do with this PR, right?

File: eZ/Publish/Core/REST/Client/Values/User/User.php [hunk 1]
```diff
@@ -77,6 +77,21 @@ public function getFieldsByLanguage($languageCode = null)
         return $this->content->getFieldsByLanguage($languageCode);
     }
 
+    /**
+     * This method returns the field for a given field definition identifier and language.
+     *
+     * If not set the initialLanguage of the content version is used.
+     *
+     * @param string $fieldDefIdentifier
+     * @param null $languageCode
```
Comment (crevillo) [inline eZ/Publish/Core/REST/Client/Values/User/User.php]: $languageCode must be always null? couldn't be other thing too?

File: eZ/Publish/Core/REST/Client/Values/User/UserGroup.php [hunk 2]
```diff
@@ -77,6 +77,21 @@ public function getFieldsByLanguage($languageCode = null)
         return $this->content->getFieldsByLanguage($languageCode);
     }
 
+    /**
+     * This method returns the field for a given field definition identifier and language.
+     *
+     * If not set the initialLanguage of the content version is used.
+     *
+     * @param string $fieldDefIdentifier
+     * @param null $languageCode
```
Comment (crevillo) [inline eZ/Publish/Core/REST/Client/Values/User/UserGroup.php]: same as above. it seems this is in all the new functions added here.

File: eZ/Publish/Core/REST/Client/Values/User/User.php (hunk 1 above)
Comment (iherak) [inline eZ/Publish/Core/REST/Client/Values/User/User.php]: Copied from original implementation in Content :)

I can change that everywhere, so it's consistent?

Comment (crevillo) [inline eZ/Publish/Core/REST/Client/Values/User/User.php]: that's probably something our ez crew friends should say, but i'd probably do another pr then to change only that.

**Comment by andrerom** [comment]
Will have a look at this next week.

**Comment by bdunogier** [comment]
I guess it makes the API a bit more consistent.

+1 for what it's worth. @andrerom ?

File: eZ/Publish/Core/REST/Client/Values/User/User.php (hunk 1 above)
Comment (andrerom) [inline eZ/Publish/Core/REST/Client/Values/User/User.php]: > I can change that everywhere, so it's consistent?

please do, it's `string|null`

**Comment by andrerom** [comment]
+1 with the phpdoc fixes @crevillo pointed out, and the failure is timing issues, happens once in a thousand, guess we should mock time :)

**Comment by iherak** [comment]
Rebased and fixed the phpdoc :)
